src/workerd/api/gpu/gpu-compute-pass-encoder.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include "gpu-bindgroup.h"
+#include "gpu-compute-pipeline.h"
+#include "gpu-query-set.h"
+#include "gpu-utils.h"
+#include <webgpu/webgpu_cpp.h>
+#include <workerd/jsg/jsg.h>
+
+namespace workerd::api::gpu {
+
+class GPUComputePassEncoder : public jsg::Object {
+public:
+  explicit GPUComputePassEncoder(wgpu::ComputePassEncoder e)
+      : encoder_(kj::mv(e)){};
+  JSG_RESOURCE_TYPE(GPUComputePassEncoder) {
+    JSG_METHOD(setPipeline);
+    JSG_METHOD(setBindGroup);
+  }
+
+private:
+  wgpu::ComputePassEncoder encoder_;
+  void setPipeline(jsg::Ref<GPUComputePipeline> pipeline);
+  void
+  setBindGroup(GPUIndex32 index, kj::Maybe<jsg::Ref<GPUBindGroup>> bindGroup,
+               jsg::Optional<kj::Array<GPUBufferDynamicOffset>> dynamicOffsets);
+  // TODO(soon): overloads don't seem to be supported
```

Comment:
Correct. To support this case you'll need to make use of `kj::OneOf` arguments. It'll be a bit... complicated.. unfortunately.